components/list/list-header.js

```diff
@@ -1,12 +1,13 @@
 import { css } from 'lit';
 import { findComposedAncestor } from '../../helpers/dom.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
 import { SelectionHeader } from '../selection/selection-header.js';
 
 /**
  * A header for list components containing select-all, etc.
  * @slot - Responsive container using `d2l-overflow-group` for `d2l-selection-action` elements
  */
-class ListHeader extends SelectionHeader {
+class ListHeader extends LocalizeCoreElement(SelectionHeader) {
```

Comment:
We shouldn't need this here I think, since the base `SelectionHeader` class already uses `LocalizeCoreElement`.

Reply:
Yeah I wasn't sure about this - in theory if we ever removed `LocalizeCoreElement` from the `SelectionHeader`, then the `localize` call in here would start silently breaking.  And we dedupe the `LocalizeMixin`, so I don't _think_ having it twice is a problem.  But in practice, it's highly unlikely we would ever refactor away the `LocalizeCoreElement` in `SelectionHeader` without the `localize` call changing too.

It feels similar to how we always want to include an import we use, even if we import another component that already imports it.  But I don't know if we've ever run into that same scenario with mixins 🤔

Reply:
Hmm... the `LocalizeCoreElement` and `LocalizeDynamicMixin` layers aren't deduped though. Currently they don't do anything that would cause problems (not referencing super anywhere) - but seems like that's its own risk of silent breakage, if either of those added something self-referencing.

Reply:
Fair enough, seems like both ways are a slight risk but the "repeated localize mixins" one is higher

Reply:
Thinking about it more, I have this super hazy memory of us already dealing with _something_ like this... @dlockhart, did you and Margaree run into something with extending `LocalizeCoreElement` multiple times?

Reply:
It had to do with combining a mixin that used `LocalizeDynamicMixin` and `LocalizeCoreElement`. We added this note to the documentation because of it:

> Note: If using LocalizeCoreElement or a mixin that utilizes LocalizeCoreElement as well as LocalizeDynamicMixin or a mixin that uses LocalizeDynamicMixin, LocalizeDynamicMixin must appear before LocalizeCoreElement in the chain.

 You might actually be in that situation, but because underneath everything it's all the same localization resources, it should be fine.

Reply:
@eKoopmans We'd mentioned passing around the langterm name instead, which I don't _love_, but I wonder if that's the best way to future-proof and divide responsibility here.  Keeps the `SelectionHeader` in charge of localizing since it knows how to do that, but still allows `ListHeader` to pick the wording

Reply:
Could we construct a unit test that ensures the localization is working correctly in `ListHeader`? That way it would catch a regression if we made a breaking change to the mixins.

Reply:
I _think_ visual-diff could catch that.

Reply:
> I think visual-diff would catch that.

Might be misunderstanding what you're saying - the label is only used for an `aria-label`, not a visual one

Reply:
I like a simple unit test, that'll also make sure it's correctly overriding the behaviour from the selection-header Something like:

```js
// Note: The property `element.ariaLabel` doesn't work in Firefox
expect(element.getAttribute('aria-label')).to.equal('Expected value');
```

Reply:
Sorry, you are correct. If it were visible though, we would want to forgo the unit test.